string map_file
int32[] type_array
---